azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,429 @@
+
+   private static int runningNumber = 1;
+   private final String azureGroup;
+
+   @Resource
+   @Named(ComputeServiceConstants.COMPUTE_LOGGER)
+   private Logger logger = Logger.NULL;
+
+   private final AzureComputeApi api;
+
+   private final AzureComputeConstants azureComputeConstants;
+
+   @Inject
+   AzureComputeServiceAdapter(final AzureComputeApi api, final AzureComputeConstants azureComputeConstants,
+                              @Named("azureGroupId") String group) {
+
+      this.api = api;
+      this.azureComputeConstants = azureComputeConstants;
+      this.azureGroup = group;
+   }
+
+   @Override
+   public NodeAndInitialCredentials<VMDeployment> createNodeWithGroupEncodedIntoName(
+           final String group, final String name, final Template template) {
+
+      final String location = template.getLocation().getId();
+      DeploymentTemplateBuilder deploymentTemplateBuilder = api.deploymentTemplateFactory().create(group, name, template);
+
+      final String loginUser = deploymentTemplateBuilder.getLoginUserUsername();
+      final String loginPassword = deploymentTemplateBuilder.getLoginPassword();
+
+      DeploymentBody deploymentTemplateBody =  deploymentTemplateBuilder.getDeploymentTemplate();
+
+      DeploymentProperties properties = DeploymentProperties.create(deploymentTemplateBody);
+
+      final String deploymentTemplate = UrlEscapers.urlFormParameterEscaper().escape(deploymentTemplateBuilder.getDeploymentTemplateJson(properties));
+
+      logger.debug("Deployment created with name: %s", name);
+
+      final Set<VMDeployment> deployments = Sets.newHashSet();
+
+      ResourceGroupApi resourceGroupApi = api.getResourceGroupApi();
+      ResourceGroup resourceGroup = resourceGroupApi.get(azureGroup);
+      final String resourceGroupName;
+
+      if (resourceGroup == null){
+         final Map<String, String> tags = ImmutableMap.of("description", "jClouds managed VMs");
+         resourceGroupName = resourceGroupApi.create(azureGroup, location, tags).name();
+      } else {
+         resourceGroupName = resourceGroup.name();
+      }
+
+      if (!retry(new Predicate<String>() {
+         @Override
+         public boolean apply(final String name) {
+            runningNumber++;
+
+            Deployment deployment = api.getDeploymentApi(resourceGroupName).create(name, deploymentTemplate);
+
+            if (deployment != null) {
+               VMDeployment vmDeployment = new VMDeployment();
+               vmDeployment.deployment = deployment;
+               deployments.add(vmDeployment);
+            } else {
+               logger.debug("Failed to create deployment!");
+            }
+            return !deployments.isEmpty();
+         }
+      }, azureComputeConstants.operationTimeout(), 1, SECONDS).apply(name)) {
+         final String illegalStateExceptionMessage = format("Deployment %s was not created within %sms so it will be destroyed.",
```

Comment:
@andreaturli The `delete` operation of a template deployment does not take care of deletion of all the resources created as part of the deployment. When you click on "Delete" from the Azure portal, this is the message displayed:

> Delete deployment
Are you sure you want to delete this deployment? This will only delete the deployment history but will leave the resources and the resource group unaffected.

From MSDN document, here is more information on what the `delete` operation does:

> A template deployment that is currently running cannot be deleted. Deleting a template deployment removes the associated deployment operations. Deleting a template deployment does not affect the state of the resource group.